### PR TITLE
Add brobnar cards

### PR DESCRIPTION
**Pull request by granttitus**
I modified Dunkoro's code from https://github.com/CorayThan/archon-arena/pull/38 and then hooked up `onPlay` effects for creatures.

### CODE REVIEW

File: src/card-scripts/ScriptUtils.ts [hunk 1]
```diff
@@ -18,15 +20,63 @@ export const friendlyCreatures = (state: GameState): Creature[] => {
...
+        }, [])
+}
+
+export const enemyCards = (state: GameState): CardInGame[] => {
+    return enemyCreatures(state).map(creature => creature as CardInGame)
+        .concat(enemyArtifacts(state).map(artifact => artifact as CardInGame))
+        .concat(enemyUpgrades(state))
+}
+
+export const friendlyPlayer = (state: GameState, card: CardInGame): PlayerState => {
```
Comment (CorayThan) [inline src/card-scripts/ScriptUtils.ts]: I'd rename this `friendlyPlayerForCard`. It had me really confused for a while. Same with its opposite.

File: src/card-scripts/ScriptUtils.ts [hunk 2]
```diff
@@ -60,3 +110,135 @@ export const putInArchives = (state: GameState, card: CardInGame, friendlyArchiv
...
+    card.ready = false
+}
+
+export const modifyAmber = (playerState: PlayerState, amount: number):number => {
+    const actualChange = playerState.amber + amount >= 0 ? amount : playerState.amber
+    playerState.amber = playerState.amber + actualChange
+    return actualChange
+}
+
+export const shuffleDeck = (state: GameState) => {
```
Comment (CorayThan) [inline src/card-scripts/ScriptUtils.ts]: will need to know which player's deck to shuffle

File: src/card-scripts/cota/brobnar/Drummernaut.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { cardScripts } from "../../CardScripts"
+import { CardScript } from "../../types/CardScript"
+import { Creature } from "../../../shared/gamestate/Creature"
+import { friendlyCreatures, putInHand } from "../../ScriptUtils"
+
+const cardScript: CardScript = {
+    //TODO would be nice to not have to duplicate this three times over, without making a dedicated function in ScriptUtils for this...
+    onPlay: {
```
Comment (CorayThan) [inline src/card-scripts/cota/brobnar/Drummernaut.ts]: Create these as a `const onPlayFightReap = { [...] }` at the top of the file and reference it in each. No copy pasting!

File: src/card-scripts/cota/brobnar/FirstBlood.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import {cardScripts} from "../../CardScripts"
+import {CardScript} from "../../types/CardScript"
+import {Creature} from "../../../shared/gamestate/Creature"
+import {House} from "../../../shared/keyforge/house/House"
+import {allCreatures, dealDamage, friendlyCreatures} from "../../ScriptUtils"
+
+const cardScript: CardScript = {
+	amber: () => 1,
+	alpha: () => true,
+    onPlay: {
```
Comment (CorayThan) [inline src/card-scripts/cota/brobnar/FirstBlood.ts]: bad spacing?

File: src/card-scripts/cota/brobnar/KelifiDragon.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import { cardScripts } from "../../CardScripts"
+import {Creature} from "../../../shared/gamestate/Creature";
+import {activePlayerState, allCreatures, dealDamage, modifyAmber} from "../../ScriptUtils"
+
+const cardScript: CardScript = {
+    power: () => 12,
+    canBePlayed: (state) => {
+        return activePlayerState(state).amber >= 7
+    },
+    fight: {
```
Comment (CorayThan) [inline src/card-scripts/cota/brobnar/KelifiDragon.ts]: same for all these, reference a const in the file

File: src/card-scripts/ScriptUtils.ts (hunk 1 above)
Comment (granttitus) [reply src/card-scripts/ScriptUtils.ts]: Done

File: src/card-scripts/ScriptUtils.ts (hunk 2 above)
Comment (granttitus) [reply src/card-scripts/ScriptUtils.ts]: We now pass in `PlayerState`

File: src/card-scripts/cota/brobnar/FirstBlood.ts (hunk 4 above)
Comment (granttitus) [reply src/card-scripts/cota/brobnar/FirstBlood.ts]: Tabs were included. Fixed

File: src/card-scripts/cota/brobnar/KelifiDragon.ts (hunk 5 above)
Comment (granttitus) [reply src/card-scripts/cota/brobnar/KelifiDragon.ts]: Done

File: src/card-scripts/cota/brobnar/Drummernaut.ts (hunk 3 above)
Comment (granttitus) [reply src/card-scripts/cota/brobnar/Drummernaut.ts]: Done

File: src/card-scripts/cota/brobnar/GrumpBuggy.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { CardScript } from "../../types/CardScript"
+import { cardScripts } from "../../CardScripts"
+import {friendlyCreatures, enemyPlayerForCard, friendlyPlayerForCard, enemyCreatures} from "../../ScriptUtils"
+
+const cardScript: CardScript = {
+    amber: () => 1,
+    staticEffect: (state, config) => {
```
Comment (granttitus) [inline src/card-scripts/cota/brobnar/GrumpBuggy.ts]: I've been changing these lines to `(state: GameState, config: CardActionConfig)` as I go.

**Pull request merged**